src/plugins/visualize/public/application/utils/get_table_columns.tsx

```diff
@@ -0,0 +1,164 @@
+            'features',
+        })}
+      />
+    );
+  } else if (item.stage === 'experimental') {
+    return (
+      <EuiBetaBadge
+        className="visListingTable__experimentalIcon"
+        label="E"
+        title={i18n.translate('visualize.listing.experimentalTitle', {
+          defaultMessage: 'Experimental',
+        })}
+        tooltipContent={i18n.translate('visualize.listing.experimentalTooltip', {
+          defaultMessage:
+            'This visualization might be changed or removed in a future release and is not subject to the support SLA.',
+        })}
+      />
+    );
+  }
+};
+
+const renderItemTypeIcon = (item: VisualizationListItem) => {
+  let icon;
+  if (item.image) {
+    icon = (
+      <img className="visListingTable__typeImage" aria-hidden="true" alt="" src={item.image} />
+    );
+  } else {
+    icon = (
+      <EuiIcon
+        className="visListingTable__typeIcon"
+        aria-hidden="true"
+        type={item.icon || 'empty'}
+        size="m"
+      />
+    );
+  }
+
+  return icon;
+};
+
+export const getTableColumns = (application: ApplicationStart, history: History) => [
+  {
+    field: 'title',
+    name: i18n.translate('visualize.listing.table.titleColumnName', {
+      defaultMessage: 'Title',
+    }),
+    sortable: true,
+    render: (field: string, { editApp, editUrl, title }: VisualizationListItem) => (
+      <EuiLink
+        onClick={() => {
+          if (editApp) {
+            application.navigateToApp(editApp, { path: editUrl });
+          } else if (editUrl) {
+            history.push(editUrl);
+          }
+        }}
+        data-test-subj={`visListingTitleLink-${title.split(' ').join('-')}`}
+      >
+        {field}
+      </EuiLink>
+    ),
+  },
+  {
+    field: 'typeTitle',
+    name: i18n.translate('visualize.listing.table.typeColumnName', {
+      defaultMessage: 'Type',
+    }),
+    sortable: true,
+    render: (field: string, record: VisualizationListItem) => (
+      <span>
+        {renderItemTypeIcon(record)}
+        {record.typeTitle}
+        {getBadge(record)}
+      </span>
+    ),
+  },
+  {
+    field: 'description',
+    name: i18n.translate('visualize.listing.table.descriptionColumnName', {
+      defaultMessage: 'Description',
+    }),
+    sortable: true,
+    render: (field: string, record: VisualizationListItem) => <span>{record.description}</span>,
+  },
+];
+
+export const getNoItemsMessage = (createItem: () => void) => (
+  <div>
+    <EuiEmptyPrompt
+      iconType="visualizeApp"
+      title={
+        <h1 id="visualizeListingHeading">
+          <FormattedMessage
+            id="visualize.listing.createNew.title"
+            defaultMessage="Create your first visualization"
+          />
+        </h1>
+      }
+      body={
+        <p>
+          <FormattedMessage
+            id="visualize.listing.createNew.description"
+            defaultMessage="You can create different visualizations, based on your data."
```

Comment:
Comma isn't necesssary
```suggestion
            defaultMessage="You can create different visualizations based on your data."
```